index.js

```diff
@@ -0,0 +1,299 @@
+(function() {
+  function Calculator(domElement) {
+    this.current = domElement,
+    this.display = this.current.querySelector('.calculator-valueDisplay')
+    this.first = 0,
+    this.second = 0,
+    this.operand = '',
+    this.result = '',
+    this.firstValueAdded = false,
+    this.secondValueAdded = false
+  }
+
+  Calculator.prototype.addTheListeners = function() {
+    var calculator = this
+    var currentCalc = this.current
+    currentCalc.addEventListener('click', function(event){
+      if(event.target.matches('.calculator-button:not(#flip):not(#percentage), #decimal')){
+        calculator.displayValue(event.target.innerHTML)
+      }else if(event.target.matches('#divide, #multiply, #subtract, #add')){
+        calculator.getOperand(event.target.innerHTML)
+      }else if(event.target.matches('#AC')){
+        calculator.allClear()
+      }else if(event.target.matches('#percentage')){
+        calculator.percent()
+      }else if(event.target.matches('#equals')){
+        calculator.calculate()
+      }else if(event.target.matches('#flip')){
+        calculator.flipSign()
+      }
+    })
+    currentCalc.onkeydown = function(event){
+      if(parseInt(event.key) || event.key == 0){
+        calculator.assignClass(event.key)
+        calculator.displayValue(event.key)
+      }
+      if(calculator.isOperand(event.key)) {
```

Comment:
I would make a hash map of all the `event.key` values that mean anything to your calculator.